File: api/app/Enums/ShopGenre.php
<?php

namespace App\Enums;

enum ShopGenre: string
{
    case RAMEN = 'ラーメン';
    case CAFE = 'カフェ';
    case IZAKAYA = '居酒屋';
    case YAKINIKU = '焼肉';
    case SUSHI = '寿司';
    case SWEETS = 'スイーツ';
    case FAST_FOOD = 'ファストフード';
    case RESTAURANT = 'レストラン';
    case BAR = 'バー';
    case BAKERY = 'ベーカリー';
    case ITALIAN = 'イタリアン';
    case CHINESE = '中華';
    case KOREAN = '韓国料理';
    case FRENCH = 'フレンチ';
    case JAPANESE = '和食';
    case WESTERN = '洋食';
    case SEAFOOD = '海鮮';
    case VEGETARIAN = 'ベジタリアン';
    case BBQ = 'BBQ';
    case OTHER = 'その他';

    /**
     * Get the display name for the genre
     */
    public function displayName(): string
    {
        return $this->value;
    }

    /**
     * Get all genre values as an array
     */
    public static function values(): array
    {
        return array_column(self::cases(), 'value');
    }

    /**
     * Get all genre names as an array
     */
    public static function names(): array
    {
        return array_column(self::cases(), 'name');
    }

    /**
     * Create genre from string value
     */
    public static function fromValue(string $value): ?self
    {
        foreach (self::cases() as $case) {
            if ($case->value === $value) {
                return $case;
            }
        }
        return null;
    }

    /**
     * Get color hex code for the genre (for admin panel consistency)
     */
    public function colorHex(): string
    {
        return match($this) {
            self::RAMEN => '#CC3333',
            self::CAFE => '#996633',
            self::IZAKAYA => '#E69900',
            self::YAKINIKU => '#B31A1A',
            self::SUSHI => '#0066CC',
            self::SWEETS => '#E666B3',
            self::FAST_FOOD => '#E6B300',
            self::RESTAURANT => '#8033B3',
            self::BAR => '#333333',
            self::BAKERY => '#CC9966',
            self::ITALIAN => '#009933',
            self::CHINESE => '#CC0000',
            self::KOREAN => '#990066',
            self::FRENCH => '#004D99',
            self::JAPANESE => '#669933',
            self::WESTERN => '#B3804D',
            self::SEAFOOD => '#00B3CC',
            self::VEGETARIAN => '#33CC33',
            self::BBQ => '#804000',
            self::OTHER => '#808080',
        };
    }

    /**
     * Check if the genre is valid
     */
    public static function isValid(string $value): bool
    {
        return in_array($value, self::values());
    }
}

File: kumatomo_ios/Domain/Entity/Enums/ShopGenre.swift
import SwiftUI

enum ShopGenre: String, CaseIterable, Codable {
    case ramen = "ラーメン"
    case cafe = "カフェ"
    case izakaya = "居酒屋"
    case yakiniku = "焼肉"
    case sushi = "寿司"
    case sweets = "スイーツ"
    case fastFood = "ファストフード"
    case restaurant = "レストラン"
    case bar = "バー"
    case bakery = "ベーカリー"
    case italian = "イタリアン"
    case chinese = "中華"
    case korean = "韓国料理"
    case french = "フレンチ"
    case japanese = "和食"
    case western = "洋食"
    case seafood = "海鮮"
    case vegetarian = "ベジタリアン"
    case bbq = "BBQ"
    case other = "その他"

    var displayName: String {
        return rawValue
    }

    var color: Color {
        switch self {
        case .ramen:
            return Color(red: 0.8, green: 0.2, blue: 0.2)
        case .cafe:
            return Color(red: 0.6, green: 0.4, blue: 0.2)
        case .izakaya:
            return Color(red: 0.9, green: 0.6, blue: 0.0)
        case .yakiniku:
            return Color(red: 0.7, green: 0.1, blue: 0.1)
        case .sushi:
            return Color(red: 0.0, green: 0.4, blue: 0.8)
        case .sweets:
            return Color(red: 0.9, green: 0.4, blue: 0.7)
        case .fastFood:
            return Color(red: 0.9, green: 0.7, blue: 0.0)
        case .restaurant:
            return Color(red: 0.5, green: 0.3, blue: 0.7)
        case .bar:
            return Color(red: 0.2, green: 0.2, blue: 0.2)
        case .bakery:
            return Color(red: 0.8, green: 0.6, blue: 0.4)
        case .italian:
            return Color(red: 0.0, green: 0.6, blue: 0.3)
        case .chinese:
            return Color(red: 0.8, green: 0.0, blue: 0.0)
        case .korean:
            return Color(red: 0.6, green: 0.0, blue: 0.4)
        case .french:
            return Color(red: 0.0, green: 0.3, blue: 0.6)
        case .japanese:
            return Color(red: 0.4, green: 0.6, blue: 0.2)
        case .western:
            return Color(red: 0.7, green: 0.5, blue: 0.3)
        case .seafood:
            return Color(red: 0.0, green: 0.7, blue: 0.8)
        case .vegetarian:
            return Color(red: 0.2, green: 0.8, blue: 0.2)
        case .bbq:
            return Color(red: 0.5, green: 0.2, blue: 0.0)
        case .other:
            return Color(red: 0.5, green: 0.5, blue: 0.5)
        }
    }

    static var allGenres: [ShopGenre] {
        return ShopGenre.allCases
    }

    static func from(string: String) -> ShopGenre? {
        return ShopGenre.allCases.first { $0.rawValue.lowercased() == string.lowercased() }
    }
}